themes/bootstrap3/js/searchbox_controls.js

```diff
@@ -18,11 +19,6 @@ VuFind.register('searchbox_controls', function SearchboxControls() {
   };
 
   function _handleInputChange(input, triggerInputEvent = true) {
-    _textInput.value = input;
```

Comment:
I think this removed code is necessary because of the virtual keyboard -- this function is not only called in response to normal input events, but also in response to user interaction with virtual keyboards. For the same reason, it may not be possible to give the reset button its own separate event handler -- it may be better to restore this code instead of setting up a separate event listener below. (In any case, note that the use of classList.toggle here is a bit more compact than the if/else in the version below).

Reply:
I like the more compact syntax but I'm in favor of keeping the listeners separate. We should only set up a listener if that part is enabled. Keeping them separate means the autocomplete, reset button, and virtual keyboard are all independent of each other instead of doing the if-statement checks below.

Reply:
Also, setting the value is redundant because we're passing the value into this function and the dispatch is redundant because this is triggered by the same event.

Reply:
Have you tested this with the virtual keyboard enabled? Maybe I'm misunderstanding something, but my impression is that the virtual keyboard does not trigger normal JS events but instead directly calls the _handleInputChange method... so if you remove the reset button logic and value setting from _handleInputChange, it will break the virtual keyboards.